src/main/scala/com/cognite/sdk/scala/common/dataTypes.scala

```diff
@@ -197,9 +197,20 @@ object EitherDecoder {
 
 sealed trait Setter[+T]
 sealed trait NonNullableSetter[+T]
-final case class SetValue[+T](set: T) extends Setter[T] with NonNullableSetter[T]
+sealed trait NonNullableIterableSetter[+T]
+final case class SetValue[+T](set: T)
+    extends Setter[T]
+    with NonNullableSetter[T]
+    with NonNullableIterableSetter[T]
 final case class SetNull[+T]() extends Setter[T]
 
+// For add/remove on updates for assets_ids, labels (array types)
+final case class AddRemoveArr[+T](add: Seq[T] = Seq.empty, remove: Seq[T] = Seq.empty)
+    extends NonNullableIterableSetter[Seq[T]]
+// For metadata add/remove on updates
+final case class AddRemoveMap(add: Map[String, String] = Map.empty, remove: Seq[String] = Seq.empty)
+    extends NonNullableIterableSetter[Map[String, String]]
```

Comment:
I'd much prefer `ChangeArray`/`ChangeSeq` and `ChangeMap`, or `UpdateArray`/`UpdateMap`.